core/domain/exp_jobs_one_off.py

```diff
@@ -74,6 +74,42 @@
 SUCCESSFUL_EXPLORATION_MIGRATION = 'Successfully migrated exploration'
 
 
+class MultipleChoiceInteractionOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that produces a list of all (exploration, state) pairs that use the
+    Multiple selection interaction and have rules that do not correspond to any
+    answer choices.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+
+        exploration = exp_fetchers.get_exploration_from_model(item)
+        for state_name, state in exploration.states.items():
+            if state.interaction.id == 'MultipleChoiceInput':
+                choices_length = len(
+                    state.interaction.customization_args['choices']['value'])
+                for anwer_group_index, answer_group in enumerate(
+                        state.interaction.answer_groups):
+                    for rule_index, rule_spec in enumerate(
+                            answer_group.rule_specs):
+                        if rule_spec.inputs['x'] > choices_length:
+                            yield (
+                                item.id,
+                                'State name:%s, AnswerGroup:%s, Rule:%s' % (
```

Comment:
add 'is invalid' as well to end of string? Also, add space after ':' and mention that the indices here are 0-indexed in the string that is yielded.

Reply:
done